meetings/2024-07-24.md

```diff
@@ -0,0 +1,177 @@
+  * Joyee (chat): If we just implement `”preload”: [“tsx”]` in package.json, it will just work out of the box. (Tsx is also a loader implementer, they implement both CJS hooks and ESM hooks)
+  * Benjamin (chat): I think when I compare user experienced the default behavior “just working” is better DX than the default behavior an error being thrown.
+  * Joyee (chat): e.g. to run TypeScript today, you already just do `node —import tsx test.ts`, and `preload` is just a on-disk config for `--import`
+  * Geoffrey (chat): I think it’s not so simple as a “preload” field in package.json (what about all the various scripts in package.json, would it apply to all of them?) but we can definitely figure out a config file solution that can work
+  * Benjamin (chat): Also I want to emphasize that we can land/iterate and see
+  * Joyee (chat): If it’s just script support, `“preload”: { “test”: [“tsx”] }` would suffice
+  * Daniel: opt-in at a package level basis.
+  * Daniel, had experimented with a loader in the past, hit a few challenges, in particular in supporting commonjs, did think about whether configurations where needed to be read. In the end why compete with the existing solutions like tsx that do a good job
+  * Antoine: can we go back to discussing Marco’s PR
+    * what would need to change for it to land?
+    * any concern with landing with a compile time flag,
+      * Marco will not likely be used?
+      * Daniel, from TypeScript team, would be a pain if it had a compile time flag.
+    * any concerns with landing with only runtime flag ?
+    * Benjamin, as long as we communicate that it is only there for experimentation. We should
+      also agree that we would not move to stable with a thumbs up from TypeScript team.
+    * Matteo, absolutely see the problems that Ryan is seeing. Happy to say npm publishing and
+      mono-repos are out of scope. But value if we can do
+  * Ryan, most concerns are around what is the next step for the users who want to publish, or
+     run in mono-repo. Don’t see paths forward for that. Which is ok. Good to set expectations if
+     some use cases are known as out of scope. If just local apps and scripts then good to
+     experiment.
+  * Jake (chat): I don't think locking it all the way behind a compile flag is really needed
+  * Daniel (chat): I think Ryan's concerns are still big. There is no built-in "grow-up" story in the current implementation when it comes to npm publishing or running in a monorepo.
+  * Benjamin (chat): Honestly at least for now - the “grow up” story is “build the code with TSC that generates .d.ts” and we likely should recommend tsc for that (we’re probably happy to have a blessed workflow in the docs)
```

Comment:
Discussion continued after that, we had consensus and blessing to land the type stripping PR as experimental and consensus to listen to and apply feedback from the TypeScript team.